examples/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,528 @@
+  expect(URL).not.toBeNull();
+  expect(URL).toBeDefined();
+  expect(authUrl).not.toBeNull();
+  expect(authUrl).toBeDefined();
+  expect(authType).not.toBeNull();
+  expect(authType).toBeDefined();
+  expect(serviceName).not.toBeNull();
+  expect(serviceName).toBeDefined();
+
+  let zoneId;
+  let zoneEtag;
+  let ruleId;
+  let ruleEtag;
+
+  test('createZone request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('createZone() result:');
+    // begin-create_zone
+
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.23.56.234',
+    };
+
+    const params = {
+      name: 'an example of zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of zone',
+    };
+
+    await contextBasedRestrictionsService
+      .createZone(params)
+      .then((res) => {
+        zoneId = res.result.id;
+        zoneEtag = res.headers.etag;
+
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-create_zone
+  });
+
+  test('listZones request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('listZones() result:');
+    // begin-list_zones
+
+    const params = {
+      accountId,
+    };
+
+    await contextBasedRestrictionsService
+      .listZones(params)
+      .then((res) => {
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-list_zones
+  });
+
+  test('getZone request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('getZone() result:');
+    // begin-get_zone
+
+    const params = {
+      zoneId,
+    };
+
+    await contextBasedRestrictionsService
+      .getZone(params)
+      .then((res) => {
+        console.log(JSON.stringify(res.result, null, 2));
+      })
+      .catch((err) => {
+        console.warn(err);
+      });
+
+    // end-get_zone
+  });
+
+  test('replaceZone request example', async (done) => {
+    consoleLogMock.mockImplementation((output) => {
+      originalLog(output);
+      done();
+    });
+    consoleWarnMock.mockImplementation((output) => {
+      done(output);
+    });
+
+    originalLog('replaceZone() result:');
+    // begin-replace_zone
+
+    // Request models needed by this operation.
+
+    // AddressIPAddress
```

Comment:
Please remove this comment and the one 2 lines above.

Reply:
removed